uint16 convoystatus
uint8[4] convoyid
